File: typecast-java/src/main/java/io/typecast/exceptions/NotFoundException.java
package io.typecast.exceptions;

/**
 * Exception thrown when the API returns a 404 Not Found error.
 * 
 * <p>This typically indicates the requested resource (e.g., voice) was not found.</p>
 */
public class NotFoundException extends TypecastException {
    
    /**
     * Creates a new NotFoundException.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     */
    public NotFoundException(String message, String responseBody) {
        super(message, 404, responseBody);
    }

    /**
     * Creates a new NotFoundException with a cause.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     * @param cause        the underlying cause
     */
    public NotFoundException(String message, String responseBody, Throwable cause) {
        super(message, 404, responseBody, cause);
    }
}

File: typecast-php/src/Exceptions/NotFoundException.php
<?php

declare(strict_types=1);

namespace Neosapience\Typecast\Exceptions;

class NotFoundException extends TypecastException
{
    public function __construct(string $message = 'Not found', ?\Throwable $previous = null)
    {
        parent::__construct($message, 404, $previous);
    }
}
